### PR TITLE
Image Classifier Python Task Library Tests

### PR DESCRIPTION
**Pull request by kinaryml**
- Changes to the Image Classifier API since the [ImageClassifierOptions](https://github.com/tensorflow/tflite-support/blob/259dd9634cb6bd799c9a10ab4e04d8358b2e381e/tensorflow_lite_support/cc/task/vision/proto/image_classifier_options.proto) proto does not have a self-contained [ClassificationOptions](https://github.com/tensorflow/tflite-support/blob/259dd9634cb6bd799c9a10ab4e04d8358b2e381e/tensorflow_lite_support/cc/task/processor/proto/classification_options.proto) proto yet. The [TextEmbedderOptions](https://github.com/tensorflow/tflite-support/blob/259dd9634cb6bd799c9a10ab4e04d8358b2e381e/tensorflow_lite_support/cc/task/text/proto/text_embedder_options.proto) proto seems to have this though
- Some fixes to tensorflow_lite_support/cc/task/vision/proto/BUILD (added a self-contained proto py proto dependency)
- Added tests for the Image Classifier API

### CODE REVIEW

File: tensorflow_lite_support/python/task/vision/pybinds/image_classifier.cc [hunk 1]
```diff
@@ -29,7 +29,7 @@ namespace {
 namespace py = ::pybind11;
 }  // namespace
 
-PYBIND11_MODULE(_pywrap_image_classifier, m) {
+PYBIND11_MODULE(image_classifier, m) {
```
Comment (khanhlvg) [inline tensorflow_lite_support/python/task/vision/pybinds/image_classifier.cc:32]: @ziyeqinghan I don't know why pybind works in OSS if being setup like this (set module name to `image_classifier`), but failed when module name is set as `_pywrap_image_classifier`.

You can run `bazel run //tensorflow_lite_support/python/test/task/vision:image_embedder_test` to reproduce that this test failed in OSS.

Comment (ziyeqinghan) [reply tensorflow_lite_support/python/task/vision/pybinds/image_classifier.cc:32]: Please change the file name `image_classifier.cc` to `_pywrap_image_classifier.cc` which should fix this.

File: tensorflow_lite_support/python/test/task/vision/image_classifier_test.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+      model_file,
+      max_results=max_results
+    )
+
+    # Loads image.
+    image = tensor_image.TensorImage.from_file(
+      test_util.get_test_data_path("burger.jpg"))
+
+    # Classifies the input.
+    image_result = classifier.classify(image, bounding_box=None)
```
[the remark this answers is not in the document]
Comment (khanhlvg) [reply tensorflow_lite_support/python/test/task/vision/image_classifier_test.py:114]: @ziyeqinghan To keep this pull request small, I asked Kinar to leave these tests for the next pull request. The tests here are just meant to serve as a convenient to quick check if the implementation is working.

You can focus on the implementation part on this PR.

File: tensorflow_lite_support/python/test/task/vision/image_classifier_test.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+
+class ImageClassifierTest(parameterized.TestCase, base_test.BaseTestCase):
+
+  def setUp(self):
+    super().setUp()
+    self.model_path = test_util.get_test_data_path(_MODEL_FLOAT)
+
+  @staticmethod
+  def create_classifier_from_options(model_file, **classification_options):
```
[the remark this answers is not in the document]
Comment (khanhlvg) [reply tensorflow_lite_support/python/test/task/vision/image_classifier_test.py:52]: ditto. More comprehensive tests will be added in the next PR.

File: tensorflow_lite_support/python/task/vision/image_classifier.py [hunk 4]
```diff
@@ -65,14 +65,7 @@ def create_from_options(cls,
     proto_options.base_options.CopyFrom(
         task_utils.ConvertToProtoBaseOptions(options.base_options))
 
-    # if options.classification_options:
```
Comment (ziyeqinghan) [inline tensorflow_lite_support/python/task/vision/image_classifier.py]: Please remain this comment and add an issue and issue link to track this TODO.

**Pull request merged**